packages/myst-cli/src/transforms/outputs.ts

```diff
@@ -47,6 +48,99 @@ export async function transformOutputsToCache(
   );
 }
 
+export function stringIsMatplotlibOutput(value?: string): boolean {
+  if (!value) return false;
+  // We can add more when we find them...
+  const match =
+    value.match(/<(Figure|Text|matplotlib)(.*)at ([0-9a-z]+)>/) ||
+    value.match(/^<(Figure|Text|AxesSubplot)(.*)>$/) ||
+    value.match(/^(Text)\((.*)\)$/);
+  return !!match;
+}
+
+/** Remove warnings and errors from outputs */
+export function transformFilterOutputStreams(
+  mdast: GenericParent,
+  vfile: VFile,
+  {
+    output_stdout: stdout = 'show',
+    output_stderr: stderr = 'show',
+    output_matplotlib_strings: mpl = 'remove-warn',
+  }: Pick<ProjectSettings, 'output_stdout' | 'output_stderr' | 'output_matplotlib_strings'> = {},
+) {
+  if (stdout === 'show' && stderr === 'show') return;
+  const blocks = selectAll('block', mdast) as GenericNode[];
+  blocks.forEach((block) => {
+    const tags: string[] = Array.isArray(block.data.tags) ? block.data.tags : [];
+    const blockRemoveStderr = tags.includes('remove-stderr');
+    const blockRemoveStdout = tags.includes('remove-stdout');
+    const outputs = selectAll('output', block) as GenericNode[];
+    // There should be only one output in the block
+    outputs.forEach((output) => {
+      output.data = output.data.filter((data: IStream | MinifiedMimeOutput) => {
+        if (stderr !== 'show' && data.output_type === 'stream' && data.name === 'stderr') {
+          const doRemove = stderr.includes('remove') || blockRemoveStderr;
+          const doWarn = stderr.includes('warn');
+          const doError = stderr.includes('error');
```

Comment:
It's a little confusing how all of these are dependent on the parts of the string value (i.e. it's not like an enum where you can swap out the actual values without changing the functionality) - especially since the values are defined all the way in `myst-frontmatter`... There's also repetition across these three filtering functions...

Potentially some of the logic could be pulled into a separate function... but hmf, it's probably fine as is, just takes a couple passes to read.

Reply:
I am not sure if this one deserves to be an enum. I agree that the code is pretty repetitive, I think I am going to leave it until another pass in this area though.

I am going to start adding an enum for the known cell tags.